designs/2024-deprecated-rule-metadata/README.md

```diff
@@ -0,0 +1,250 @@
+- Repo: eslint/eslint
+- Start Date: 2024-02-20
+- RFC PR: <https://github.com/eslint/rfcs/pull/116>
+- Authors: [bmish](https://github.com/bmish)
+
+# Support additional metadata for rule deprecations
+
+## Summary
+
+<!-- One-paragraph explanation of the feature. -->
+
+This RFC suggests a format for storing additional information in rule metadata about rule deprecations and replacement rules, allowing automated documentation and website tooling to generate more informative deprecation notices.
+
+## Motivation
+
+<!-- Why are we doing this? What use cases does it support? What is the expected
+outcome? -->
+
+There are long-time rule [properties](https://eslint.org/docs/latest/extend/custom-rules#rule-structure) `meta.deprecated` and `meta.replacedBy` that have been intended to document when rules are deprecated and what their replacement rule(s) are. For the most part, usage would look something like this:
```

Comment:
Small accessibility note: links sometimes get read without their context. Having informative link descriptions/titles helps clarify them. https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html

```suggestion
There are long-time [rule properties](https://eslint.org/docs/latest/extend/custom-rules#rule-structure) `meta.deprecated` and `meta.replacedBy` that have been intended to document when rules are deprecated and what their replacement rule(s) are. For the most part, usage would look something like this:
```

Reply:
Thank you for the tip, will try to keep this in mind for my links.